File: scripts/python/03_json.py
#!/usr/bin/env python3
"""JSON mode - request structured output and parse it."""
import json
import re

from openai import OpenAI

client = OpenAI(base_url="http://localhost:11434/v1", api_key="not-needed")

response = client.chat.completions.create(
    model="apple-foundationmodel",
    messages=[
        {
            "role": "user",
            "content": (
                "Return JSON with fields 'chip', 'year', 'cores'. "
                "Describe the Apple M1 chip. Return ONLY JSON."
            ),
        },
    ],
    response_format={"type": "json_object"},
    max_tokens=120,
)

raw = (response.choices[0].message.content or "").strip()
raw = re.sub(r"^```(?:json)?\s*|\s*```$", "", raw, flags=re.MULTILINE).strip()
data = json.loads(raw)
print(json.dumps(data, indent=2, sort_keys=True))

File: scripts/nodejs/03_json.mjs
// JSON mode - request structured output and parse it.
import OpenAI from "openai";

const client = new OpenAI({ baseURL: "http://localhost:11434/v1", apiKey: "not-needed" });

const response = await client.chat.completions.create({
  model: "apple-foundationmodel",
  messages: [
    {
      role: "user",
      content:
        "Return JSON with fields 'chip', 'year', 'cores'. " +
        "Describe the Apple M1 chip. Return ONLY JSON.",
    },
  ],
  response_format: { type: "json_object" },
  max_tokens: 120,
});

let raw = (response.choices[0].message.content || "").trim();
raw = raw.replace(/^```(?:json)?\s*|\s*```$/gm, "").trim();
const data = JSON.parse(raw);
console.log(JSON.stringify(data, null, 2));
